### ISSUE TITLE
Enhancement: Add 'secondsElapsed' param to update()

### ISSUE THREAD

**Issue by richpixel** [asaia/StarlingPunk#2]
Hi- I've been playing around with the Engine. The simple & clean design is just what I'm looking for! Nice job.

For the future, I think it would be a good idea to pass a param like 'secondsElapsed' to most of the update() functions every frame. This is somewhat standard - if the fps takes a hit for whatever reason, you can update your entities based on a 'velocity per second' instead of a fixed rate per frame.

So basically the SPEntity.update() function would receive a param, 'secondsElapsed' which is a Number - the number of seconds elapsed since last frame, usually something like 0.04.

-rich

**Comment by asaia** [comment]
Thanks glad you're finding it useful! I completely agree with you-- I'll add some kinda of DeltaTime param to the next release.

**Comment by smithmj5** [comment]
First off, cool framework, really liking it!

I'm trying to tell by looking at the code, but finding the answer a bit elusive - StarlingPunk looks to be running on a fixed timestep, is that correct?  And if so, what happens if the framerate drops?  I know that FlashPunk has an option for fixed vs. variable timestep, and I noticed that comment is still in your code above the constructor for SPEngine, but there's no actual code for it.  It's "@param   fixed           If a fixed-framerate should be used." but that's not anywhere in the engine.  Any plans for offering both at some point?

**Comment by richpixel** [comment]
Yes it's a fixed time step. I requested a variable time-step in my message and he said it would be in the next release.

**Comment by smithmj5** [comment]
Sorry, I read his reply as saying he'd simply pass an elapsed time to each update function, not change the whole engine to actually be a variable framerate. My bad. Cause in a fixed framerate, the elapsed time should always be the same...it's been a long day lol

**Comment by richpixel** [comment]
Oh sorry- maybe I misread what you wrote! Well the engine uses the ENTER_FRAME event for time-steps so that is always fixed by whatever framerate you set on the stage. The elapsed time param is to get the true amount of time a frame took to process- to deal with cases when it takes longer than the defined framerate.

**Comment by asaia** [comment]
Correct, the engine currently uses Starling's ENTER_FRAME event, and I'm planning on exposing a passedTime variable so that you can ensure your characters run at the same speed even if the frame-rate drops. I didn't have any plans to include a variable time-step. Animations and Tweens are handled by Starling's Juggler class which wouldn't be effected by a variable time-step anyways. I'm just going to trust Starling's internal code to know when and how to advance frames. Unless anyone can think of a use case where Starlings magic wouldn't work.

**Comment by asaia** [comment]
added passedTime property to SP class
